include/CBawFile.h

```diff
@@ -77,9 +78,31 @@ namespace core
 		void* getData() { return this; }
 	};
 
+	template<typename T>
+	struct VariableSizeBlob : Blob {};
```

Comment:
"curiously recurring template" will allow us to declare static calcBlobSizeForObj and allocMemForBlob functions which can be "overriden" like virtual functions

Reply:
yea maye.. But what's the point of doing so?

Reply:
not having to remember to specialize the allocMemForBlob function every time :)

Reply:
The function is not specialized. It's not even a template, only struct is a template. See how it's used [in mesh writer](https://github.com/buildaworldnet/IrrlichtBAW/pull/21/files#diff-69beb7cce857086a5dc3a282bae8004aR65)

Reply:
Sorry I meant declared and defined for every struct.

Reply:
'fixed'